utils/R/convert.input.R

```diff
@@ -22,73 +23,193 @@ convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, st
+      ## Or running non met2model conversion scripts (download.raw, met2cf, etc.)
+      
+      existing.dbfile <- dbfile.input.check(siteid = site.id,
+                                            mimetype = mimetype, 
+                                            formatname = formatname, 
+                                            parentid = input.id, 
+                                            startdate = start_date,
+                                            enddate = end_date, 
+                                            con = con, 
+                                            hostname = host$name)
+      
+      
+      print(existing.dbfile, digits = 10)
+      print("end CHECK")
+      
+      
+      if (nrow(existing.dbfile) > 0) {
+        
+        existing.input <- db.query(paste0("SELECT * FROM inputs WHERE id=", existing.dbfile[["container_id"]]),con)
+        
+        # Convert dates to Date objects and strip all time zones
+        # (DB values are timezone-free)
+        start_date <- lubridate::force_tz(lubridate::as_date(start_date), "UTC")
+        end_date   <- lubridate::force_tz(lubridate::as_date(end_date), "UTC")
+        
+        existing.input$start_date <- lubridate::force_tz(lubridate::as_date(existing.input$start_date), "UTC")
+        existing.input$end_date   <- lubridate::force_tz(lubridate::as_date(existing.input$end_date), "UTC")
+        
+        if (overwrite) {
+                        # collect files to flag for deletion
+          
+                        files.to.delete <- remote.execute.R( paste0("list.files('",
+                                                                    existing.dbfile[["file_path"]],
+                                                                    "', full.names=TRUE)"),
+                                                                     host, user = NA, verbose = TRUE,R = "R")
+          
+                        file.deletion.commands <- .get.file.deletion.commands(files.to.delete)
+          
+                        remote.execute.R( file.deletion.commands$move.to.tmp,
+                                          host, user = NA, 
+                                          verbose = TRUE,R = "R")
+          
+                        # Schedule files to be replaced or deleted on exiting the function
+                        successful <- FALSE
+                        on.exit(if (exists("successful") && successful) {
+            
+                        logger.info("Conversion successful, with overwrite=TRUE. Deleting old files.")
+                        remote.execute.R( file.deletion.commands$delete.tmp, host, user = NA, verbose = TRUE,  R = "R" )
+            
+                      } else {
+                          logger.info("Conversion failed. Replacing old files.")
+                          remote.execute.R(file.deletion.commands$replace.from.tmp, host, user = NA, verbose = TRUE, R = "R" )
+          })
```

Comment:
Also, I think I follow the flow of your logic as to why this needed to be here, but this block looks identical to the overwrite block above and it would be nice to avoid having the same code twice. Leave for now, but think about it.